RIOT-OS/RIOT
# gnrc_lorawan: add support for class C

### Contribution description
This PR adds support for Class C in GNRC LoRaWAN.
To activate, enable the `CONFIG_GNRC_LORAWAN_CLASS_C` CFLAG.

With Class C LoRaWAN nodes keep the receiver on all the time and therefore show the lowest downlink latency.
Class C is also supported by common LoRaWAN network servers (e.g TTN, Chirpstack).



### Testing procedure
- Enable Class C support in the LoRaWAN Network Server (e.g TTN)
- Run `examples/gnrc_lorawan` with `CFLAGS=-DCONFIG_GNRC_LORAWAN_CLASS_C`. Join the network and transmit and uplink.
- Schedule a downlink to the LoRaWAN node. The node should indicate packet reception.
Also, make sure that LoRaWAN class A still works as expected.



### Issues/PRs references
Depends on #19342
